### ISSUE TITLE
Memory leak present in window management

### ISSUE THREAD

**Issue by dallbee** [raysan5/raylib#358]
When trying to debug #357, I discovered that there appears to be a memory leak in raylib's core window code.

I've included a minimal sample and a valgrind dump to reproduce leak. It's small (24 bytes).

```
Compiler: gcc 7.2.0
OpenGL version string: 4.5.0 NVIDIA 384.69
Platform: x86_64 Linux
CPU Model: Intel(R) Core(TM) i5-4670K CPU @ 3.40GHz
Linux version: 4.12.13-1-ARCH
Raylib: 1.7.0
```

```c
#include <raylib.h>


int main() {
    InitWindow(1024, 720, "Test");
    CloseWindow();
    return 0;
}
```

Valgrind Dump:
```
dylan:arch [develop ±] ➜  valgrind ./client --leak-check=full -vvv                ~/P/B/b/meson-out
==26399== Memcheck, a memory error detector
==26399== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==26399== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==26399== Command: ./client --leak-check=full -vvv
==26399== 
INFO: Initializing raylib (v1.7.0)
INFO: Display device initialized successfully
INFO: Display size: 1600 x 900
INFO: Render size: 1024 x 720
INFO: Screen size: 1024 x 720
INFO: Viewport offsets: 0, 0
INFO: GLAD: OpenGL extensions loaded successfully
INFO: OpenGL 3.3 Core profile supported
INFO: GPU: Vendor:   NVIDIA Corporation
INFO: GPU: Renderer: GeForce GTX 760/PCIe/SSE2
INFO: GPU: Version:  3.3.0 NVIDIA 384.69
INFO: GPU: GLSL:     3.30 NVIDIA via Cg compiler
INFO: Number of supported extensions: 330
INFO: [EXTENSION] DXT compressed textures supported
INFO: [EXTENSION] ETC2/EAC compressed textures supported
INFO: [EXTENSION] Anisotropic textures filtering supported (max: 16X)
INFO: [EXTENSION] Clamp mirror wrap texture mode supported
INFO: [TEX ID 1] Texture created successfully (1x1)
INFO: [TEX ID 1] Base white texture loaded successfully
INFO: [VSHDR ID 1] Vertex shader compiled successfully
INFO: [FSHDR ID 2] Fragment shader compiled successfully
INFO: [SHDR ID 3] Shader program loaded successfully
INFO: [SHDR ID 3] Default shader loaded successfully
INFO: [CPU] Default buffers initialized successfully (lines, triangles, quads)
INFO: [VAO ID 1] Default buffers VAO initialized successfully (lines)
INFO: [VAO ID 2] Default buffers VAO initialized successfully (triangles)
INFO: [VAO ID 3] Default buffers VAO initialized successfully (quads)
INFO: OpenGL default states initialized successfully
INFO: [TEX ID 2] Texture created successfully (128x128)
INFO: [TEX ID 2] Default font loaded successfully
INFO: [TEX ID 2] Unloaded texture data from VRAM (GPU)
INFO: [TEX ID 1] Unloaded texture data (base white texture) from VRAM
INFO: Window closed successfully
==26399== 
==26399== HEAP SUMMARY:
==26399==     in use at exit: 76,591 bytes in 392 blocks
==26399==   total heap usage: 10,004 allocs, 9,612 frees, 164,390,566 bytes allocated
==26399== 
==26399== LEAK SUMMARY:
==26399==    definitely lost: 24 bytes in 1 blocks
==26399==    indirectly lost: 0 bytes in 0 blocks
==26399==      possibly lost: 0 bytes in 0 blocks
==26399==    still reachable: 76,567 bytes in 391 blocks
==26399==         suppressed: 0 bytes in 0 blocks
==26399== Rerun with --leak-check=full to see details of leaked memory
==26399== 
==26399== For counts of detected and suppressed errors, rerun with: -v
==26399== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

```

**Comment by raysan5** [comment]
raylib uses internally [GLFW3](https://github.com/glfw/glfw) for windows management, maybe this leak is related to glfw?

Could you provide more info about the leaked block?

**Comment by dallbee** [comment]
@raysan5 I just tested this out on the develop branch and there appears to be no leak. Wherever it was, you fixed it in 1.8.

Again, sorry for not checking develop, and thank you very much for your help. 👍

**Comment by raysan5** [comment]
👍
